### PR TITLE
JA 10 - Think about removing JA2BETAVERSION code

### PR DESCRIPTION
**Pull request by edwbuck**
This series of commits demonstrates what the ja2-stracciatella code would look like with JA 10 implemented.  

After removing the JA2BETAVERSION pre-processor directive, it was obvious that some of the code
was misleading, in the sense that empty function calls existed within the code base.  For clarity, each
of the empty functions were removed, with a separate commit (for ease of review).

Also, after removing the JA2BETAVERSION pre-processor directive, it was obvious that one function 
returned a constant BOOLEAN value of false.  Refactoring was done to remove unreachable code 
blocks and to simplify the program logic in light of the function's inability to return True.  For clarity, this
was done in a separate commit for ease of review.

Finally, a number pre-processor directives lost the ability to be set dynamically.  Previously, these mapped function names to implementations that only existed when JA2BETAVERSION was enabled.  When not enabled, they mapped to ((void)0) which the compiler would optimize out as an expression 
statement with  no side-effect.  For clarity, these pre-processor directives were removed, as was the
text that expanded to the NOOP expression.  Again, a separate commit was done for ease of review.

Please consider this code submission for the closing of #10 .  I believe the code will be easier to read and maintain without the misleading text. 

Thank you.
Edwin Buck

### CODE REVIEW

**Comment by lynxlynxlynx** [comment]
LGTM. Maybe the cheats could be salvaged — some sound potentially useful for development purposes.

& wow, welcome back Edwin! One of the first porters to gcc and make.

File: src/game/SaveLoadScreen.cc [hunk 1]
```diff
@@ -48,10 +48,6 @@
 #include "Meanwhile.h"
 #include "PreBattle_Interface.h"
 
-#if defined JA2BETAVERSION
```
Comment (derSteFfi) [inline src/game/SaveLoadScreen.cc]: this leaves a newline in line 50 that's unneccessary

File: src/game/Strategic/Auto_Resolve.cc [hunk 2]
```diff
@@ -71,10 +71,6 @@
 #include "slog/slog.h"
 #include "Easings.h"
 
-#ifdef JA2BETAVERSION
```
Comment (derSteFfi) [inline src/game/Strategic/Auto_Resolve.cc]: again, empty newline here.

File: src/game/Strategic/MapScreen.cc [hunk 3]
```diff
@@ -273,9 +273,6 @@ enum MapEvent
 	MAP_EVENT_PLOT_PATH,
 	MAP_EVENT_CANCEL_PATH,
```
Comment (derSteFfi) [inline src/game/Strategic/MapScreen.cc]: again, empty newline

File: src/game/Tactical/Soldier_Init_List.cc [hunk 4]
```diff
@@ -33,12 +33,6 @@
 #include "MemMan.h"
 #include "FileMan.h"
```
Comment (derSteFfi) [inline src/game/Tactical/Soldier_Init_List.cc]: here again :)

File: src/game/Tactical/Soldier_Init_List.h [hunk 5]
```diff
@@ -70,10 +70,6 @@ void AddProfilesNotUsingProfileInsertionData(void);
 
 void StripEnemyDetailedPlacementsIfSectorWasPlayerLiberated();
```
Comment (derSteFfi) [inline src/game/Tactical/Soldier_Init_List.h]: and here

File: src/game/Tactical/Turn_Based_Input.cc [hunk 6]
```diff
@@ -2413,12 +2395,6 @@ void GetKeyboardInput(UIEventKind* const puiNewEvent)
 			}
 			else if ( InputEvent.usKeyState & CTRL_DOWN )
       {
```
Comment (derSteFfi) [inline src/game/Tactical/Turn_Based_Input.cc]: this leaves the else if empty and ready for removal

File: src/game/Tactical/World_Items.cc [hunk 7]
```diff
@@ -23,10 +23,6 @@
 #include "MemMan.h"
 #include "FileMan.h"
```
Comment (derSteFfi) [inline src/game/Tactical/World_Items.cc]: here is a newline again

File: src/game/TacticalAI/NPC.h [hunk 8]
```diff
@@ -96,10 +96,6 @@ BOOLEAN NPCWillingToAcceptItem( UINT8 ubNPC, UINT8 ubMerc, OBJECTTYPE * pObj );
 void SaveBackupNPCInfoToSaveGameFile(HWFILE);
 void LoadBackupNPCInfoFromSavedGameFile(HWFILE);
```
Comment (derSteFfi) [inline src/game/TacticalAI/NPC.h]: newline :)

File: src/game/SaveLoadScreen.cc (hunk 1 above)
Comment (edwbuck) [inline src/game/SaveLoadScreen.cc]: Don't mind fixing white space issues.  Look for another commit removing them.

File: src/game/Strategic/Auto_Resolve.cc (hunk 2 above)
Comment (edwbuck) [inline src/game/Strategic/Auto_Resolve.cc]: Will be fixed in next commit on JA-10 branch

File: src/game/Strategic/MapScreen.cc (hunk 3 above)
Comment (edwbuck) [inline src/game/Strategic/MapScreen.cc]: Will be fixed in next commit on JA-10 branch.

File: src/game/Tactical/Soldier_Init_List.cc (hunk 4 above)
Comment (edwbuck) [inline src/game/Tactical/Soldier_Init_List.cc]: Will be fixd in next commit on JA-10 branch.

File: src/game/Tactical/Soldier_Init_List.h (hunk 5 above)
Comment (edwbuck) [inline src/game/Tactical/Soldier_Init_List.h]: Fixed in next commit on JA2 branch.

File: src/game/Tactical/Turn_Based_Input.cc (hunk 6 above)
Comment (edwbuck) [inline src/game/Tactical/Turn_Based_Input.cc]: Removal won't be trivial to review, or implement.  Cutting the block puts more keystrokes in the else block which appears three lines below (due to poor formatting and poor newline / brace style choices, it appears to be an else block for a different else statement).

I don't mind cleaning up the flock-of-seagulls nesting smell, but I haven't determined if I'll be doing that in this branch.  Currently I'm leaning against it, but if it's the only item left, please consider merging without repair in this case.

The empty block is the safest, even if it is less clear.

File: src/game/Tactical/World_Items.cc (hunk 7 above)
Comment (edwbuck) [inline src/game/Tactical/World_Items.cc]: Fixed in next commit on JA2 branch.

File: src/game/TacticalAI/NPC.h (hunk 8 above)
Comment (edwbuck) [inline src/game/TacticalAI/NPC.h]: Fixed in next commit on JA2 branch.

**Comment by edwbuck** [comment]
Lynxlynxlynx,

Thank you for the kind words.  It is heartwarming to know that a few people remember me, even from programming work done a decade or so ago.

In a few short words, I cannot summarize all that's happened in the last decade.  I hope that time has treated all my JA2 programming friends well.  Life is very different for me now, and in general I'm enjoying it.

Cheers,
Edwin

**Comment by edwbuck** [comment]
I wouldn't worry about salvaging the cheats until we can implement them with polymorphisim.  Currently we lack the policy object to be replaced with the cheating policy.  The structure of cheating by "if statement" makes the code harder to read.

File: src/game/Tactical/Turn_Based_Input.cc (hunk 6 above)
Comment (edwbuck) [inline src/game/Tactical/Turn_Based_Input.cc]: I'm open to putting this as a bug on the backlog, and merging what's done into the mainline.  That way we don't lose the task, and we can shut the branch down quickly.

Comment (derSteFfi) [inline src/game/Tactical/Turn_Based_Input.cc]: ah, ok. I didn't notice the tabbing here. In this case I agree, let's solve that somewhere else.

**Comment by lynxlynxlynx** [comment]
It's easy to remember, since I think I was the only one helping you back then, with most players being on windows. Looking at the old wiki, I see I wrote a bunch of regexy fixers for includes and leading whitespace (again a problem). Old days of CVS. :)

Thanks for you work!

**Pull request merged**